lldb/source/API/SBCommandInterpreter.cpp

```diff
@@ -571,6 +571,15 @@ SBStructuredData SBCommandInterpreter::GetStatistics() {
   return data;
 }
 
+SBStructuredData SBCommandInterpreter::GetTranscript() {
+  LLDB_INSTRUMENT_VA(this);
+
+  SBStructuredData data;
+  if (IsValid())
+    data.m_impl_up->SetObjectSP(m_opaque_ptr->GetTranscript());
```

Comment:
Here don't want to hand out a shared pointer to the internal transcript. If we do and the caller starts accessing it, we can crash if new commands are added to the structured data. If we to not copy things, one thing we could do it to grab the transript shared pointer from the command interpreter and have it create a new one.

Reply:
Made a (deep) copy in the latest commit. Added test to verify that the user's copy isn't modified when more commands are run in the debugger.